server/src/custom_logging.py

```diff
@@ -0,0 +1,7 @@
+import logging
+
+
+def setup_logging():
+    # OPTIMIZE: this will result in different indents for different log levels, but avoiding this is tricky
```

Comment:
What about: https://stackoverflow.com/a/62802973/4089458 ?

Reply:
i don't know how to make that work in this example, can you spell it out? it looks like here the string needs to be known in advance?

Reply:
Something like this I'd say. The offset can be a bit shorter. Say we take the longest levelname and add 2 to it

```python3
>>> format = lambda x,y: f"{'[' + x + ']':<10} {y}"
>>> format("ERROR","foo bar")
'[ERROR]    foo bar'
>>> format("INFO","foo bar")
'[INFO]     foo bar'
```

Reply:
but how do you replace this code

```
def setup_logging():
    # OPTIMIZE: this will result in different indents for different log levels, but avoiding this is tricky
    log_format = "%(levelname)s:     %(message)s"
    logging.basicConfig(level=logging.INFO, format=log_format)
```

with that code? I need to provide a string in the `format` parameter.